### DISCUSSION TITLE
Show More for Infinite Hits

### DISCUSSION THREAD

**Discussion by avremel** [algolia/instantsearch#6081] [Q&A] [answered]
Algolia makes a `multipleQueries` request with a) the primary query, b) another query for each facet which has a selection ([similar to this Solr feature](url)).

When using [InfiniteHits](https://www.algolia.com/doc/api-reference/widgets/infinite-hits/react/#hook-api-showmore) and the user hits Show More, the `multipleQueries` request includes all the facet queries.

Why are those facet queries necessary if all I care about is getting the next page of hits?

**Comment by Haroenv** [comment, marked as answer]
In theory those requests aren't needed, however the design of InstantSearch right now is that every change does a request and we only render based on the results of the last request. infinite hits is an exception to that and still reads from previous queries through the cache.

Right now I don't know how we could do those infinite hits (or pagination, as that's really the same issue) queries without re-requesting the faceting information, and this optimisation on network time may cost more in complexity of the code. I'm not sure of course, and if you have a good idea on how to contribute, we definitely welcome that!

Note that the pricing of Algolia is per network request, not per query, so pricing-wise this won't make a difference.

**Reply by avremel** [reply]
The reason I'm asking is because I'm exploring rewriting our list page without `instantsearch` (but with the `algoliasearch` client). I want to make sure my assumptions are correct.

By the way: When you refer to the cache are you referring to the cache builtin to the `algoliasearch` client?

**Reply by Haroenv** [reply]
infinite hits has its own cache internally: https://github.com/algolia/instantsearch/blob/master/packages/instantsearch.js/src/connectors/infinite-hits/connectInfiniteHits.ts#L206
